staging/src/k8s.io/apiserver/pkg/audit/request.go

```diff
@@ -233,3 +238,13 @@ func LogAnnotations(ae *auditinternal.Event, annotations map[string]string) {
 		LogAnnotation(ae, key, value)
 	}
 }
+
+// truncate User-Agent if too long, otherwise return it directly.
+func maybeTruncateUserAgent(req *http.Request) string {
+	ua := req.UserAgent()
+	if len(ua) > maxUserAgentLength {
+		ua = string([]byte(ua)[:maxUserAgentLength]) + userAgentTruncateSuffix
```

Comment:
You don't need to cast - just `ua[:maxUserAgentLength]` should work.

Reply:
ok